### ISSUE TITLE
Erratic behaviors when installing pytorch3d from github on colab

### ISSUE THREAD

**Issue by WxComeon** [facebookresearch/pytorch3d#462]
I followed your instruction to install the stable version of PyTorch3d on github using the following block in colab,
```
import os
import sys
import torch
if torch.__version__=='1.6.0+cu101' and sys.platform.startswith('linux'):
    !pip install pytorch3d
else:
    !pip install 'git+https://github.com/facebookresearch/pytorch3d.git@stable'
```
Installation was fast two weeks ago. However, 
1) now it often stuck on building wheel for pytorch3d (setup.py) and then threw an error:
```
Building wheel for pytorch3d (setup.py) ... error
  ERROR: Failed building wheel for pytorch3d
  Running setup.py clean for pytorch3d
  Building wheel for fvcore (setup.py) ... done
  Created wheel for fvcore: filename=fvcore-0.1.2.post20201122-cp36-none-any.whl size=44456 sha256=cff8a2c89859bc29ecb5bd2f1ff17e3431fc7870e304f758ef61b56c21771711
  Stored in directory: /root/.cache/pip/wheels/33/7b/ec/59dffd0a0aa995739aad60ac97e70252247e9d17eae69c1cee
  Building wheel for pyyaml (setup.py) ... done
  Created wheel for pyyaml: filename=PyYAML-5.3.1-cp36-cp36m-linux_x86_64.whl size=44619 sha256=f4fddc2dfc297fafe5ab93f0d9b31c942c198685022be76200b531c79b7de656
  Stored in directory: /root/.cache/pip/wheels/a7/c1/ea/cf5bd31012e735dc1dfea3131a2d5eae7978b251083d6247bd
Successfully built fvcore pyyaml
Failed to build pytorch3d
Installing collected packages: pyyaml, yacs, portalocker, fvcore, pytorch3d
  Found existing installation: PyYAML 3.13
    Uninstalling PyYAML-3.13:
      Successfully uninstalled PyYAML-3.13
    Running setup.py install for pytorch3d ... error
ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-zv7y1e2_/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-zv7y1e2_/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-1gf8a59u/install-record.txt --single-version-externally-managed --compile Check the logs for full command output.
```
2) Even when I successfully installed pytorch3d (very slow, took from 20 to 30 minutes), if I want to import pytorch3d.structures or any module in it:
```
from torchvision.utils import save_image
from pytorch3d.structures import Meshes
from pytorch3d.transforms import euler_angles_to_matrix, Transform3d, Rotate, Translate, Scale
```
It says module not found:
```
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-16-1550c16ae9c3> in <module>()
      5 from torchvision.utils import save_image
      6 import pytorch3d
----> 7 from pytorch3d.structures import Meshes
      8 from pytorch3d.transforms import euler_angles_to_matrix, Transform3d, Rotate, Translate, Scale
      9 from pytorch3d.renderer import (

ModuleNotFoundError: No module named 'pytorch3d.structures'
```

The behaviors are very random. Please help. Thank you very much!

**Comment by bottler** [comment]
A couple of things have changed in the last two weeks, both of which are relevant.

1) On 11 November there was a new release of pytorch3d. The stable tag moved. PyTorch3D now has an additional build-time dependency - NVidia CUB, and the tutorials were changed.

The cell for importing / installing pytorch3d in all the notebooks is now

```
!pip install torch torchvision
import os
import sys
import torch
if torch.__version__=='1.6.0+cu101' and sys.platform.startswith('linux'):
    !pip install pytorch3d
else:
    need_pytorch3d=False
    try:
        import pytorch3d
    except ModuleNotFoundError:
        need_pytorch3d=True
    if need_pytorch3d:
        !curl -LO https://github.com/NVIDIA/cub/archive/1.10.0.tar.gz
        !tar xzf 1.10.0.tar.gz
        os.environ["CUB_HOME"] = os.getcwd() + "/cub-1.10.0"
        !pip install 'git+https://github.com/facebookresearch/pytorch3d.git@stable'
```

The important thing here is that CUB gets downloaded, expanded, and has its location stored in an environment variable, all before the same install-from-source command.

2) Colab changed its built-in version of torch from 1.6.0 to 1.7.0. I don't know the date. We have prebuilt wheels on PyPI for pytorch 1.6.0 (because we wanted to match colab), which can be installed really quickly with `pip install pytorch3d` if the torch version is correct.

Basically the installation is slow because of point 2) - the cell is detecting that it needs to build pytorch3d from source because the PyTorch version is wrong. It then fails because of point 1). If you add the `curl`, `tar` and `os.environ` commands to your notebook things should work, but I'm afraid they will still be slow.

**Comment by WxComeon** [comment]
@bottler Hey Jeremy, thank you so much for the quick response! I downgraded PyTorch to 1.6.0 and then used the code you provided in colab, and now it is very fast! 

For those who also want to downgrade PyTorch:
```
!python3 -c "import torch;assert torch.__version__.startswith('1.6'), 'should be 1.6.x'" || pip install torch==1.6.0+cu101 torchvision==0.7.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html --upgrade 
```

**Comment by bottler** [comment]
Thanks for sharing that command - it could well be useful to others.
